src/types/Todo.ts

```diff
@@ -0,0 +1,16 @@
+import { User } from './User';
+
+export interface TodoWithoutUser {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+export interface TodoWithUser {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User | null;
+}
```

Comment:
Why do you need two types like this?

I would suggest you make the user optional
Even if you have some reasons to make it null rather than undefined, you should extend interfaces one from another (remember - DRY)